src/backend/utils/fmgr/dfmgr.c

```diff
@@ -297,6 +298,13 @@ internal_load_library(const char *libname)
 		file_tail = file_scanner;
 	}
 
+	if (file_exists("data_encryption.key") && file_read_buffer_modify_hook == NULL)
```

Comment:
Isn't there a better place for this check? Can this be done in the postmaster start, right before loading data files. Wasn't `internal_load_library()` be called when loading any so files? This doesn't look like the right place for the hook.

And, please add a test in TDE side to ensure the behaviour.